pkg/testing/ess/serverless_provision.go

```diff
@@ -71,114 +62,118 @@ func (prov *ServerlessProvision) SetLogger(l runner.Logger) {
 	prov.log = l
 }
 
-// Provision a new set of serverless instances
-func (prov *ServerlessProvision) Provision(ctx context.Context, requests []runner.StackRequest) ([]runner.Stack, error) {
-	upWaiter := sync.WaitGroup{}
-	depErrs := make(chan error, len(requests))
-	depUp := make(chan bool, len(requests))
-	stacks := []runner.Stack{}
-	for _, req := range requests {
-		client := NewServerlessClient(prov.cfg.Region, "observability", prov.cfg.APIKey, prov.log)
-		srvReq := ServerlessRequest{Name: req.ID, RegionID: prov.cfg.Region}
-		proj, err := client.DeployStack(ctx, srvReq)
-		if err != nil {
-			return nil, fmt.Errorf("error deploying stack for request %s: %w", req.ID, err)
-		}
-		err = client.WaitForEndpoints(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("error waiting for endpoints to become available for request: %w", err)
-		}
-		newStack := runner.Stack{
-			ID:            req.ID,
-			Version:       req.Version,
-			Elasticsearch: client.proj.Endpoints.Elasticsearch,
-			Kibana:        client.proj.Endpoints.Kibana,
-			Username:      client.proj.Credentials.Username,
-			Password:      client.proj.Credentials.Password,
-			Internal: map[string]interface{}{
-				"deployment_id":   proj.ID,
-				"deployment_type": proj.Type,
-			},
-		}
-		stacks = append(stacks, newStack)
-		prov.stacksMut.Lock()
-		prov.stacks[req.ID] = stackhandlerData{client: client, stackData: newStack}
-		prov.stacksMut.Unlock()
+// Create creates a stack.
+func (prov *ServerlessProvision) Create(ctx context.Context, request runner.StackRequest) (runner.Stack, error) {
+	// allow up to 4 minutes for requests
+	createCtx, createCancel := context.WithTimeout(ctx, 4*time.Minute)
+	defer createCancel()
 
-		upWaiter.Add(1)
-		go func() {
-			isUp, err := client.DeploymentIsReady(ctx)
-			if err != nil {
-				depErrs <- err
+	client := NewServerlessClient(prov.cfg.Region, "observability", prov.cfg.APIKey, prov.log)
+	srvReq := ServerlessRequest{Name: request.ID, RegionID: prov.cfg.Region}
 
-			}
-			depUp <- isUp
-		}()
+	prov.log.Logf("Creating serverless stack %s (%s)", request.Version, request.ID)
+	proj, err := client.DeployStack(createCtx, srvReq)
+	if err != nil {
+		return runner.Stack{}, fmt.Errorf("error deploying stack for request %s: %w", request.ID, err)
+	}
+	err = client.WaitForEndpoints(createCtx)
+	if err != nil {
+		return runner.Stack{}, fmt.Errorf("error waiting for endpoints to become available for request: %w", err)
```

Comment:
Did you miss the request id?
```suggestion
		return runner.Stack{}, fmt.Errorf("error waiting for endpoints to become available for request %s: %w", request.ID, err)
```

Reply:
+1

Reply:
Updated.

Reply:
are you sure you updated the right line? it's still the same, but it isn't a blocker so it's fine